liveRecorder/Tasks/ConcatinationTask.py

```diff
@@ -124,6 +133,100 @@ def get_flavor_id(self, url_postfix, single_flavor):
 
         return flavor_id
 
+    def is_processing_required(self):
+        # read metadata file && get entry server nodes to decide whether to process the job or skip it
+        should_process = True
+        local_dc_type_str = 'UNKNOWN'
+        remote_dc_type_str = 'UNKNOWN'
+        try:
+            with open(self.metadata_full_path, "r") as dc_file:
```

Comment:
this function is TOO long... also the logic can be simplified....  just do a generic try except...
also there is a log of code there that is useless like local_dc_type_str or remote_dc_type_str

Reply:
took outside all the business logic